helix-core/src/main/java/org/apache/helix/ConfigAccessor.java

```diff
@@ -87,26 +88,7 @@
    * @param builder
    */
   private ConfigAccessor(Builder builder) {
```

Comment:
I tried to compare this solution with the other method, a private constructor accepts all parameters including the ZkClient.
Pros: less parameter.
Cons: you have to make createZkClient() method public.

Since the constructor is private and easy to update, will the option B be better? Moreover, many classes have it's defined constructors so you don't need to define a new one.
In this way, the builder will be very clean. Basically, the only public methods are setXXX() and build(). All the other methods are not required to be public. What do you think?

Reply:
In order to not expose createZkClient(), I'll make that have protected access and modify the constructors.